baseplate/sidecars/secrets_fetcher.py

```diff
@@ -332,6 +340,11 @@ def fetch_secrets(
     return soonest_expiration
 
 
+def trigger_callback(callback: Optional[str], secrets_file: str) -> None:
+    if callback:
+        subprocess.Popen([callback, secrets_file])
```

Comment:
I've got a few concerns about the external script in failure scenarios, not sure how many of these matter:

* if the script takes a while to run, its output will be interleaved with the fetcher daemon's. I do think we want its logs to go to the same place for debugging, but I'm not really sure of a cleaner way here. Probably just a note and not actionable.
* if the script takes a while to run and secrets are fetched again, we might end up with multiple of the external script piled up. should we only fire the callback if the callback isn't already running? or say that's up to the end user to care about?

Reply:
Yeah #2 could be bad. `subprocess.Popen` gives us the pid, so we can check if it's still running on the next run and log some message if we skip it. How frequently does the fetcher typically run?